src/computation/computation.resolver.ts

```diff
@@ -1,18 +1,69 @@
-import { HttpService } from '@nestjs/common'
+import * as grpc from '@grpc/grpc-js'
+import * as protoloader from '@grpc/proto-loader'
+import { Logger } from '@nestjs/common'
+import { ConfigService } from '@nestjs/config'
 import { Args, Query, Resolver } from '@nestjs/graphql'
-import { CourseEntry, CourseEntryInput } from 'src/graphql'
+import {
+  CourseRecommendationRequest,
+  CourseRecommendationResponse,
+} from 'src/graphql'
 
 @Resolver('Computation')
 export class ComputationResolver {
-  constructor(readonly computationClient: HttpService) {}
+  private metadata: grpc.Metadata
 
-  @Query('recommendCourses')
-  async recommendCourses(
-    @Args('selectedCourses') selectedCourses: CourseEntryInput[]
-  ): Promise<CourseEntry[]> {
-    const resp = await this.computationClient
-      .post('/api/suggest/courses', { selectedCourses })
-      .toPromise()
-    return resp.data.suggestions
+  private courseRecommendation: any
```

Comment:
do we really have to use `any` here?

Reply:
Yes. Typescript support for grpc is really bad